lib/iris/cube.py

```diff
@@ -2278,8 +2282,68 @@ def _intersect_modulus(self, coord, minimum, maximum, min_inclusive,
                 # because we've already restricted the coordinate's
                 # range to its modulus).
                 end_of_first_chunk = non_zero_step_indices[0]
-                subsets = [slice(inside_indices[end_of_first_chunk + 1], None),
-                           slice(None, inside_indices[end_of_first_chunk] + 1)]
+
+                # Ensure that we do not wrap if blocks are at the very edge.
+                # That is, if the very edge if wrapped and corresponds to
+                # base + period, stop this unnecessary wraparound.
+                # NOTE: The relationship between final index wrapping and
+                # being an increasing coordinate; and first index and being
+                # a decreasing coordinate, is based on the behaviour of
+                # wrap_lons.  wrap_lons wraps points at base + period.  How
+                # they are wrapped always identifies the direction of the
+                # coordinate.
+
+                # The following condition holds for increasing or decreasing
+                # coordinates with wraparound.
+                # Condition1: The two blocks don't themselves wrap
+                #             (inside_indices is contiguous).
+                # Condition2: Are we chunked at either extreme edge.
+                index_of_second_chunk = inside_indices[end_of_first_chunk + 1]
+                final_index = points.size - 1
+                if ((index_of_second_chunk ==
+                        inside_indices[end_of_first_chunk] + 1) and
+                        index_of_second_chunk in (final_index, 1)):
+
+                    # Check if final index is wrapped (increasing coordinate).
+                    # i.e. start of the second chunk is at the final index.
+                    if index_of_second_chunk == final_index:
+                        # Unwrap points and bounds
+                        if (bounds is not None and
+                                np.isclose(coord.bounds[-1, -1],
+                                           coord.bounds[0, 0] + modulus)):
+                            bounds[-1, :] = coord.bounds[-1, :]
+                            points[-1] = coord.points[-1]
+                            subsets = [slice(inside_indices[0],
+                                             inside_indices[-1] + 1)]
+                        # Unwrap points
+                        elif np.isclose(coord.points[-1],
+                                        coord.points[0] + modulus):
+                            points[-1] = coord.points[-1]
+                            subsets = [slice(inside_indices[0],
+                                       inside_indices[-1] + 1)]
```

Comment:
I managed to write a test for this that was not actually testing it - I have corrected this now.
Out of interest how did you spot this?  Did you but break points here like I just did?